terraform-docker/apply/main.yaml

```diff
@@ -52,6 +61,55 @@ stages:
                         artifact: "${{ env.name }}.enc"
                         path: $(Build.Repository.LocalPath)/${{ parameters.terraformFolder }}/.terraform/plans
 
+                    - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+                      displayName: "Get AWS credentials"
+                      inputs:
+                        workingDirectory: "."
+                        awsCredentials: ${{ format(parameters.awsServiceConnectionTemplate, env.name) }}
+                        regionName: ${{ parameters.awsRegion }}
+                        scriptType: inline
+                        inlineScript: |
+                          set -ex
+                          AWS_CONFIG_FILE=${HOME}/.aws/config
+                          AWS_SHARED_CREDENTIALS_FILE=${HOME}/.aws/credentials
+                          TMPFILE=${HOME}/.aws/tmp.json
+                          mkdir $(dirname $TMPFILE)
+                          aws sts get-federation-token --name xkf-$ENV --policy-arns arn="$AWS_ARN" --region $AWS_REGION > $TMPFILE
```

Comment:
Notice the ability to define your own policy ARN.
--polcy-arns support multiple arn commands, if we have a need we could add multiple arns but for now I don't see the need.

I wrote this special in the docs but I can understand that many security admins will look at this and say it's really dumb to give the developers the option to define the arn policy they want to use. It's a fair point but you can never go over the access that you have in the service connection that you are using to run the pipeline.

So it's up to system administrators to lock down the system that way.
Also I can't come up with a another way since else we would have to hard code the policy arn in the template and i definitely don't want to do that.